purchases/src/test/java/com/revenuecat/purchases/attributes/SubscriberAttributesDeviceCacheTests.kt

```diff
@@ -163,46 +180,249 @@ class SubscriberAttributesDeviceCacheTests {
 
         underTest.setAttributes(appUserID, mapOf(expectedAttributes.keys.toList()[1] to expectedAttributes.values.toList()[1]))
 
-        assertCapturedEqualsExpected(expectedAttributes)
+        assertCapturedEqualsExpected(mapOf(appUserID to expectedAttributes))
     }
 
     @Test
     fun `clearing caches also clears the subscriber attributes`() {
-        underTest.clearCachesForAppUserID()
-        verify { mockEditor.remove(underTest.subscriberAttributesCacheKey("appUserID")) }
+        val expectedAttributes = mapOf(
+            "tshirtsize" to SubscriberAttribute("tshirtsize", "L", isSynced = true)
+        )
+        mockNotEmptyCache(expectedAttributes)
+        underTest.clearCachesForAppUserID(appUserID)
+        verify {
+            mockEditor.putString(
+                "com.revenuecat.purchases.$apiKey.subscriberAttributes",
+                JSONObject().also {
+                    it.put("attributes", JSONObject())
+                }.toString()
+            )
+        }
+    }
+
+    @Test
+    fun `Given there are some unsynced attributes for multiple users`() {
+        val attributes = mapOf(
+            "tshirtsize" to SubscriberAttribute("tshirtsize", "L"),
+            "age" to SubscriberAttribute("age", "L", isSynced = true)
+        )
+        mockNotEmptyCacheMultipleUsers(mapOf(
+            appUserID to attributes,
+            "user2" to attributes
+        ))
+        val unsyncedSubscriberAttributes = underTest.getUnsyncedSubscriberAttributes()
+        assertThat(unsyncedSubscriberAttributes.size).isEqualTo(2)
+        unsyncedSubscriberAttributes.forEach { (_, subscriberAttributesForUser) ->
+            assertThat(subscriberAttributesForUser.size).isEqualTo(1)
+            assertThat(subscriberAttributesForUser["tshirtsize"]).isEqualTo(attributes["tshirtsize"])
+        }
+    }
+
+    @Test
+    fun `Given there are some unsynced attributes for one of multiple users`() {
+        val expectedAttribute = SubscriberAttribute("tshirtsize", "L")
+        mockNotEmptyCacheMultipleUsers(mapOf(
+            appUserID to  mapOf(
+                "tshirtsize" to expectedAttribute,
+                "age" to SubscriberAttribute("age", "L", isSynced = true)
+            ),
+            "user2" to  mapOf(
+                "tshirtsize" to SubscriberAttribute("tshirtsize", "L", isSynced = true),
+                "age" to SubscriberAttribute("age", "L", isSynced = true)
+            )
+        ))
+        val unsyncedSubscriberAttributes = underTest.getUnsyncedSubscriberAttributes()
+        assertThat(unsyncedSubscriberAttributes.size).isEqualTo(1)
+        unsyncedSubscriberAttributes.forEach { (_, subscriberAttributesForUser) ->
+            assertThat(subscriberAttributesForUser.size).isEqualTo(1)
```

Comment:
should maybe also check that the userID is `appUserID`

Reply:
bump on this comment

Reply:
It was addressed with this change https://github.com/RevenueCat/purchases-android/pull/135/commits/5ae04fd8e3cd029e8cb3a5436e6876a88ec9b4c8#diff-df42f37caebf0be122ebf14478d90b06R238